smart-contracts/contracts/airdrop/src/helpers.rs

```diff
@@ -0,0 +1,275 @@
+};
+
+use crate::state::{AirdropConfig, AIRDROP_CONFIG, REPLY_MAP, USER_INFO};
+use crate::AirdropErrors;
+
+/// Checks if the sender is the contract admin. Returns an error if not authorized.
+///
+/// # Arguments
+///
+/// * `querier` - QuerierWrapper to query contract admin information.
+/// * `env` - Environment information.
+/// * `sus_admin` - Address of the sender.
+///
+/// # Returns
+///
+/// Returns `Ok(())` if the sender is authorized as the contract admin, otherwise returns an Unauthorized error.
+pub fn is_contract_admin(
+    querier: &QuerierWrapper,
+    env: &Env,
+    sus_admin: &Addr,
+) -> Result<(), AirdropErrors> {
+    // Get the contract admin address from the contract's information
+    let contract_admin = querier
+        .query_wasm_contract_info(&env.contract.address)?
+        .admin;
+
+    // Check if the contract admin address exists
+    if let Some(contract_admin) = contract_admin {
+        // Compare the contract admin address with the provided sus_admin address
+        if contract_admin != *sus_admin {
+            // If they don't match, return an Unauthorized error
+            return Err(AirdropErrors::Unauthorized {});
+        }
+    } else {
+        // If the contract admin address doesn't exist, return an Unauthorized error
+        return Err(AirdropErrors::Unauthorized {});
+    }
+
+    // If all checks pass, return Ok() to indicate success
+    Ok(())
+}
+
+/// Adds a reply to the storage and returns a SubMsg containing the reply.
+///
+/// # Arguments
+///
+/// * `storage` - Mutable storage to save the reply mapping.
+/// * `msg` - CosmosMsg to be used as a reply.
+/// * `user` - Address of the user to associate with the reply.
+///
+/// # Returns
+///
+/// Returns a SubMsg containing the reply message and the associated reply ID.
+pub fn add_reply(
+    storage: &mut dyn Storage,
+    msg: CosmosMsg,
+    user: Addr,
+) -> Result<SubMsg, AirdropErrors> {
+    let last = REPLY_MAP
+        .range(storage, None, None, Order::Descending)
+        .next();
+    let mut id: u64 = 0;
+    if let Some(val) = last {
+        id = val?.0 + 1;
+    }
+    REPLY_MAP.save(storage, id, &user.to_string())?;
+
+    Ok(SubMsg::reply_on_success(msg, id))
+}
+
+/// Checks if the total claimable amount exceeds the airdrop amount.
+///
+/// # Arguments
+///
+/// * `total_in_user_info` - Total claimable amount from all users.
+/// * `current_airdrop_amount` - Current airdrop amount in the contract.
+///
+/// # Returns
+///
+/// Returns a default response if the total claimable amount does not exceed the airdrop amount,
+/// otherwise returns an error indicating insufficient funds.
+pub fn check_amounts_and_airdrop_size(
+    total_in_user_info: Uint128,
+    current_airdrop_amount: Uint128,
+) -> Result<Response, AirdropErrors> {
+    // Check if the total claimable amount exceeds the airdrop amount
+    if total_in_user_info > current_airdrop_amount {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
```

Comment:
This should be a variant of `AirdropErrors`

Reply:
Sure.